android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java

```diff
@@ -242,6 +268,23 @@ public static void copyStartUpSo() {
         } else {
           WXEnvironment.extractSo();
         }
+
+        Closeable w = null;
+        try {
+          if(!versionFile.exists())
+            versionFile.createNewFile();
+          FileWriter fileWriter = new FileWriter(versionFile);
+          fileWriter.write(String.valueOf(WXEnvironment.CORE_JSB_SO_VERSION));
```

Comment:
If line 277 crashes, the file still be open.

         FileWriter x = null
         try{
                x= new FileWriter
          }finally{
                if(x != null) x.close()
          }